src/test/scala/org/broadinstitute/dsde/firecloud/EntityServiceSpec.scala

```diff
@@ -387,6 +393,196 @@ class EntityServiceSpec extends BaseServiceSpec with BeforeAndAfterEach {
+        actual shouldBe a [FireCloudExceptionWithErrorReport]
+        val fex = actual.asInstanceOf[FireCloudExceptionWithErrorReport]
+        fex.errorReport.statusCode should contain (StatusCodes.ImATeapot)
+        fex.errorReport.message shouldBe "Import Service unit test intentional error"
+      }
+    }
+    "should return Import Service's error if cWDS errors and Import Service errors" in {
+      val jobId = UUID.randomUUID().toString
+
+      // set up mocks
+      val cwdsDAO = mockito[MockCwdsDAO]
+      val importServiceDAO = mockito[MockImportServiceDAO]
+
+      when(cwdsDAO.isEnabled).thenReturn(true)
+      doThrow(new ApiException(500, "cWDS unit test intentional error"))
+        .when(cwdsDAO).getJobV1(any[String], ArgumentMatchers.eq(jobId))(any[UserInfo])
+      when(importServiceDAO.getJob(any[String], any[String], ArgumentMatchers.eq(jobId))(any[UserInfo]))
+        .thenReturn(Future.failed(
+          new FireCloudExceptionWithErrorReport(
+            ErrorReport(
+              StatusCodes.ImATeapot,
+              "Import Service unit test intentional error"))
+        ))
+
+      // inject mocks to entity service
+      val entityService = getEntityService(mockImportServiceDAO = importServiceDAO, cwdsDAO = cwdsDAO)
+
+      // get job via entity service
+      val getJobFuture = entityService.getJob("workspaceNamespace", "workspaceName", jobId, dummyUserInfo("mytoken"))
+
+      ScalaFutures.whenReady(getJobFuture.failed) { actual =>
+        actual shouldBe a [FireCloudExceptionWithErrorReport]
+        val fex = actual.asInstanceOf[FireCloudExceptionWithErrorReport]
+        fex.errorReport.statusCode should contain (StatusCodes.ImATeapot)
+        fex.errorReport.message shouldBe "Import Service unit test intentional error"
+      }
+    }
+    "should not call cWDS or Rawls if cWDS is not enabled" in {
+      val jobId = UUID.randomUUID().toString
+
+      val importServiceResponse = ImportServiceListResponse(jobId, "status1", "filename1", None)
+
+      // set up mocks
+      val cwdsDAO = mockito[MockCwdsDAO]
+      val importServiceDAO = mockito[MockImportServiceDAO]
+      val mockRawlsDAO = mockito[MockRawlsDAO]
+
+      when(cwdsDAO.isEnabled).thenReturn(false)
+      when(importServiceDAO.getJob(any[String], any[String], ArgumentMatchers.eq(jobId))(any[UserInfo]))
+        .thenReturn(Future.successful(importServiceResponse))
+
+      // inject mocks to entity service
+      val entityService = getEntityService(mockImportServiceDAO = importServiceDAO, cwdsDAO = cwdsDAO, rawlsDAO = mockRawlsDAO)
+
+      // get job via entity service
+      val actual = entityService.getJob("workspaceNamespace", "workspaceName", jobId, dummyUserInfo("mytoken")).futureValue
+
+      actual shouldBe importServiceResponse
+      verify(cwdsDAO, never).getJobV1(any[String], any[String])(any[UserInfo])
+      verify(mockRawlsDAO, never).getWorkspace(any[String], any[String])(any[UserInfo])
+    }
+  }
+
```

Comment:
I tried extracting the implementation for these tests into a reusable function and thus avoiding all the copy/paste. There's enough variation in the tests that the reusable function became quite heavy with business logic and conditional branching; I reverted and left the tests fully written out like this.